commit 7c1a338154172d9cd4a0b9da60b5550e235ff29b
Author: Peter Maydell <peter.maydell@linaro.org>
Date:   Thu Sep 12 11:17:51 2013 +0100

    device_tree.c: Terminate the empty reservemap in create_device_tree()
    
    Device trees created with create_device_tree() may not have any
    entries in their reservemap, because the FDT API requires that the
    reservemap is completed before any FDT nodes are added, and
    create_device_tree() itself creates a node.  However we were not
    calling fdt_finish_reservemap(), which meant that there was no
    terminator in the reservemap list and whatever happened to be at the
    start of the FDT data section would end up being interpreted as
    reservemap entries.  Avoid this by calling fdt_finish_reservemap()
    to add the terminator.
    
    Signed-off-by: Peter Maydell <peter.maydell@linaro.org>
    Acked-by: Alexander Graf <agraf@suse.de>

Index: qemu/device_tree.c
===================================================================
--- qemu.orig/device_tree.c	2013-12-18 12:23:58.227375603 -0600
+++ qemu/device_tree.c	2013-12-18 12:23:58.215375603 -0600
@@ -41,6 +41,10 @@
     if (ret < 0) {
         goto fail;
     }
+    ret = fdt_finish_reservemap(fdt);
+    if (ret < 0) {
+        goto fail;
+    }
     ret = fdt_begin_node(fdt, "");
     if (ret < 0) {
         goto fail;
